django/db/migrations/operations/models.py

```diff
@@ -449,10 +494,11 @@ def reduce(self, operation, app_label=None):
 
 class FieldRelatedOptionOperation(ModelOptionOperation):
     def reduce(self, operation, app_label=None):
-        if (isinstance(operation, FieldOperation) and
-                self.name_lower == operation.model_name_lower and
-                not self.references_field(operation.model_name, operation.name)):
-            return [operation, self]
+        if isinstance(operation, FieldOperation) and self.name_lower == operation.model_name_lower:
+            if isinstance(operation, RemoveField):
+                return False
+            if not self.references_field(operation.model_name, operation.name):
+                return [operation, self]
```

Comment:
This `reduce()` method could actually be completely removed at this point (no tests fail if it's removed) because the `CreateModel`/`FieldRelatedOptionOperation` reduction implemented in the second commit renders it unnecessary. I could remove it in a following commit.